javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/MethodResolutionLogic.java

```diff
@@ -337,21 +340,13 @@ public static boolean isApplicable(MethodUsage method, String name, List<Resolve
         }
         return true;
     }
-
+    
     /**
-     * Filters out duplicate {@param methods} by their signature.
+     * Filters by given function {@param keyExtractor}.
```

Comment:
It is still not clear what this function does.
Why should I want to add something to a set then I then throw away?

Reply:
Please check https://stackoverflow.com/questions/23699371/java-8-distinct-by-property. Its a stateful filter mechanism.